packages/react-native-views/src/components/CameraView/index.jsx

```diff
@@ -34,187 +59,81 @@ export default function CameraView({
+  // Use fake activity when you need to render ActivityIndicator for better UX
+  const [fakeActivity, handleFakeActivity] = useFakeActivity();
+
+  // Wraps taken pictures with Sights sights prop and metadata
+  const picturesWrapper = usePictures(camera, sights, onTakePicture, handleFakeActivity);
+  const { activeSight, handleTakePicture, pictures } = picturesWrapper;
+
+  // Data payload given for common user callbacks
+  const payload = useMemo(
+    () => ({ pictures, camera, sights }),
+    [camera, pictures, sights],
+  );
+
+  // Wraps states and callbacks to manage UI in one hook place
+  const ui = useUI(camera, pictures, onCloseCamera, onShowAdvice);
+  const { height, width } = ui.container.measures;
+  const ratio = useMemo(() => makeRatio(width, height), [height, width]);
+
+  // When last picture is taken
+  useSuccess(onSuccess, payload, handleFakeActivity);
 
   return (
     <View style={styles.root}>
       <StatusBar hidden />
-
+      {/* container */}
       <SafeAreaView>
         <View
           style={styles.container}
-          onLayout={(e) => {
-            const layout = e.nativeEvent.layout;
-            setMeasures({
-              // shortest to be height always
-              height: Math.min(layout.width, layout.height),
-              // longest to be width always
-              width: Math.max(layout.width, layout.height),
-            });
-          }}
+          onLayout={ui.container.handleLayout}
         >
-          {/* pictures scroll preview sidebar */}
-          <Components.PicturesScrollPreview
-            activeSight={activeSight}
-            sights={sights}
-            pictures={pictures}
-            ref={scrollRef}
-          />
-
-          {/* camera and mask overlay */}
-          <View>
-            {measures.width ? (
-              <Components.Camera
-                onCameraReady={handleCameraReady}
-                ratio={makeRatio(measures.width, measures.height)}
-              />
-            ) : null}
-            <View style={styles.overLaps}>
-              {fakeActivity && <ActivityIndicatorView />}
-              {(!fakeActivity && camera) && (
-                <Components.Mask
-                  resizeMode="contain"
-                  id={activeSight.id}
-                  width="100%"
-                  style={styles.mask}
+          <>
+            {/* pictures scroll preview sidebar */}
+            <CameraScrollView
+              activeSight={activeSight}
+              pictures={pictures}
+              sights={sights}
+            />
+
+            {/* camera and mask overlay */}
+            <View>
+              {ui.container.measures.width && (
+                <Components.Camera
+                  onCameraReady={handleCameraReady}
+                  ratio={ratio}
                 />
               )}
+              <CameraOverlay
+                activeSightId={activeSight.id}
+                camera={camera}
+                fakeActivity={Boolean(fakeActivity)}
```

Comment:
Why not make sure it's a boolean in the `useFakeActivity` hook directly?

Reply:
Because fakeActivity is a timeoutId and if we want to break the activity, we can juste use `clearTimeout(fakeActivity)`.